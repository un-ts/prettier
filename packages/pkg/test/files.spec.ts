import prettier from 'prettier'

import PkgPlugin from 'prettier-plugin-pkg'

test('files', () => {
  const fixture = {
    files: ['lib', 'rules', '!lib/*.tsbuildinfo', 'bin'],
  }

  const input = JSON.stringify(fixture, null, 2)
  const output = prettier.format(input, {
    filepath: 'package.json',
    parser: 'json-stringify',
    plugins: [PkgPlugin],
    pluginSearchDirs: false,
  })

  expect(output).toMatchSnapshot()
})
