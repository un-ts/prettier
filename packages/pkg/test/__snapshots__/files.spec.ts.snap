// Vitest Snapshot v1

exports[`files 1`] = `
"{
  \\"files\\": [
    \\"bin\\",
    \\"lib\\",
    \\"rules\\",
    \\"!lib/*.tsbuildinfo\\"
  ]
}
"
`;
